### DISCUSSION TITLE
Does Allure Report Azure DevOps extension support attributes for better tests organization

### DISCUSSION THREAD

**Discussion by ciconq** [allure-framework/allure2#2526] [Questions & Support]
Hello, 

Just wanted to understand if this is a feature to be implemented or it should already work.

Currently I am using `NUnit` and `Allure.NUnit`, both latest versions but when adding for example `[AllureSuite("Test")]` to my test class I cannot see any changes in the report that is generated in `Azure DevOps` but `locally` everything is working as intended.

Maybe this is a bug but I am not sure as for example previously the report `Test body` was working fine in the `extension` but for a couple of weeks it is unstructured and the `attached screenshots` are not displayed(using `AllureApi`).

Also I have tested the attributes with the obsolete imports(`NUnit.Allure.Attributes`) and the new one(`Allure.NUnit.Attributes`) but the result is the same - nothing is changed and I can see the namespace for suite name instead the custom one that I have added.

**Comment by darkosusic** [comment]
Can confirm the same issue. Everything works locally, but it does not seem the attributes API is working when using the Allure extension in Azure DevOps pipelines. I'm just getting the plain report, with namespaces and test method names.

Would be great to get some answer on this.

**Comment by delatrie** [comment]
Hi, @darkosusic, @ciconq. I'm trying to look into this. Could you please share a code example?
Does the low-level API fails to work too?

```c#
AllureLifecycle.Instance.UpdateTestCase(t =>
{
    t.labels.Add(new(){
        name = LabelName.SUITE,
        value = "foo"
    });
});
```

**Reply by ciconq** [reply]
Hey, @delatrie. I have tried this with a couple of very basic tests just to see if the report would be generated as expected in Azure DevOps.

From what I tried today I can tell that the attributes are working(at least tested with `AllureSuite`) but was with the `DotNetCoreCLI@2` task(it was working when I created this thread as well).

The problem is that I am using `VSTest@2` task which is not working. The only temporary solution for the moment is to use the custom batch size option in `VSTest@2` task, if it is not specified explicitly, the attributes are not working.

I have also tried your suggestion to use the low-level API but still the same result. 
Have tried with both `Allure` version `2.26.0` and latest `2.30.0` and `Allure.NUnit` nuget package version `2.11.0` and latest `2.12.1`.

I think that this [bug](https://github.com/orgs/allure-framework/discussions/2537) might be related(behavior is described in my main post here but I haven't checked if it is working in later versions although I doubt it as I think the problem described here would be fixed as well).

**Comment by darkosusic** [comment]
Hey everyone!

Just to touch base, as I haven’t had the time to commit to this. I’ll start working on it again next week, so expect additional questions.

Thanks for replying!

I’ll get back to ya.
